File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    li = list(s)
    stk = []
    for i in li:
        if i == "(":
            stk.append(i)
            continue
            
        # i == ")"
        if not stk or stk[-1] == ")":
            return False
        elif stk[-1] == "(":
            stk.pop()
            
        
    return len(stk) == 0
        

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let stk = []
    
    for(let v of s) {
        if (v === ")" && stk.at(-1) === "(") {
            stk.pop();continue;
        }
        
        stk.push(v);
    }
    
    return stk.length === 0 ? true : false;

}